inc/cachify.class.php

```diff
@@ -1124,6 +1086,80 @@ private static function _is_logged_in() {
 		return false;
 	}
 
+	/**
+	 * Register all hooks to flush the total cache
+	 *
+	 * @since   2.4.0  Make the function public
+	 *
+	 * @param   void
+	 * @return  void
+	 */
+	private static function register_flush_cache_hooks(){
+
+		/* Define all default flush cache hooks */
+		$flush_cache_hooks = array(
```

Comment:
One more thing: I would perhaps suggest to refactor `$flush_cache_hooks` array in `action` => `priority` fashion:
```php
$flush_cache_hooks = array(
    'cachify_flush_cache' => 10,
    '_core_updated_successfully' => 10,
    'switch_theme' => 10,
    ...
);
```

This way it is much easier to alter/remove hooks from the array when necessary, no for-loop or array_map/array_filter is necessary:
```php
add_filter( 'cachify_flush_cache_hooks', function ( $hooks ) {
    // Do not flush when post is trashed.
    unset( $hooks[ 'wp_trash_post' ] );
    // Change priority, flush later (for whatever reason).
    $hooks[ 'user_register' ] = 20;

    return $hooks;
});
```

Reply:
Really good point!

Reply:
updated the PR and included the recommended changes